### ISSUE TITLE
Installing Zotero as a non-admin user

### ISSUE THREAD

**Issue by maaliso** [retorquere/zotero-deb#96]
Hi,

I have no admin previliges and i want to install this debian version on ubuntu. 
1. I have a problem with running the first line in the installation instruction, i.e., 
`wget -qO- https://raw.githubusercontent.com/retorquere/zotero-deb/master/install.sh | sudo bash`
It results in the terminal asking for my user password and a message saying that i am not allowed to run this kind of command. I tried same command (the one in the quotes) again without `sudo`, but it is also not working and the terminal asks again for the user password and tells me i not allowed to run this command.

2. I tried removing the O from the options of the wget and wrote the following code
`wget -q https://raw.githubusercontent.com/retorquere/zotero-deb/master/install.sh | bash`
This results in 2 folders called .git and .github being created in my directory and upon running the `sudo apt install zotero`, a counter runs to 100%, but then the terminal says "E: Unable to locate package zotero".

What can I do to install the software then using this version? Trying another version from the main website is not working fully because of the broken simlink to the the ./local/share/applications path. This problem is, however, out of the scope of this issue.

**Comment by retorquere** [comment]
I'd seriously recommend not experimenting this way. You don't want to send unvetted output through `bash`.

> This results in 2 folders called .git and .github being created in my directory

It does not. Those directories must have been there already; `wget -q` used this way downloads the file `install.sh` to the working directory but provides no console output, so you're sending an empty string through `bash`, which (fortunately in this case) means it does precisely nothing.

It is technically [sort of possible](https://www.baeldung.com/linux/install-packages-no-root-privileges) to install .debs without root, but there's no benefit over just unpacking the tarball Zotero provides, and will have the added downside that the installation from the `.deb` has Zotero's own update mechanism disabled since you would be expected to let `apt` handle upgrades. Without `apt` in the loop, you would have to manually check for updates.

**Comment by retorquere** [comment]
TL;DR: follow the Linux instructions from [Zotero](https://www.zotero.org/support/installation).

> As you can see in the available folders from [this page,](https://github.com/retorquere/zotero-deb) there is no folder called .git. 

I'm quite familiar with the structure of this repo and git usage.

> It is created during the process of trying to unsuccessfully install zotero.

That may well be, but that's another claim entirely. What you have done in the process of trying to install Zotero is unclear to me, but if you have these two folders, it is clearly more than just the two commands you highlighted above.

The `.git` folder is created if you clone https://github.com/retorquere/zotero-deb.git using `git`, or if you download and unpack https://github.com/retorquere/zotero-deb/archive/refs/heads/master.zip, none of which you should be using to install Zotero (and none of which is even hinted at in the install process); this repository contains the code to *package* the Zotero program, it does *not* contain the Zotero program. A process runs [every two hours](https://github.com/retorquere/zotero-deb/actions) to download Zotero from [Zotero itself](https://www.zotero.org/download/), and package it if a newer version was found. Nothing in this repo except `install.sh` is useful for people installing Zotero.

The `wget` call does not create these directories. Neither of the two commands you highlighted above will, when executed in an empty directory, create folders `.git` and/or `.github`. If you want to verify, create an empty folder, `cd` into it, then issue the two commands above. The first will error out without doing anything as you don't have root access, the second will silently "succeed" in downloading the file `install.sh` in the previously empty directory, and nothing else. This `install.sh` is of no use to you if you don't have root access.

> So, as far as i understand, there is no way of installing Zotero from this git version without admin privilege.

You don't install from this "git version" at all. If you use the .debs, you install, in the end, from https://zotero.retorque.re/file/apt-package-archive, which is where my debs actually live; this repo has the packaging code and the script to set your system up to use the packages (the one-time `install.sh`, which in itself doesn't even install Zotero on your system). You could in principle download Zotero from there, but you shouldn't, as that has automatic updates turned off. If you don't have root access, you're better off with the [official tarball](https://www.zotero.org/download/). If you have no root access, these packaged Zotero debs are not useful.

> Am I correct? If I understood wrongly, and there is actually a method, could you please highlight it?

Without root access, you should use the official Zotero tarball.

> If possible, please state directly the solutions specific to this example and not general solutions about how to handle .deb files. I would pretty much appreciate it, since i am totally novice with linux and ubuntu.

And I'm not, so when I say that the commands you issued do not create the `.git` and `.github`, you can trust me or verify.

But you should either get admin access, or use the tarball from Zotero.

**Comment by maaliso** [comment]
Ok. Thanks for your answer. Detailed and informative
